protocol/synthetix/contracts/storage/Pool.sol

```diff
@@ -372,6 +385,65 @@ library Pool {
         return self.vaults[collateralType].consolidateAccountDebt(accountId);
     }
 
+    /**
+     * @dev Allocates rewards sent at the pool level to vaults so they can be claimed
+     */
+    function updateRewardsToVaults(
+        Data storage self,
+        Vault.PositionSelector memory pos
+    ) internal returns (uint256[] memory rewards, address[] memory distributors) {
+        bytes32[] memory poolRewardIds = new bytes32[](self.rewardIds.length());
+
+        {
+            bytes32 actorId = pos.collateralType.toBytes32();
+            uint256 totalSharesD18 = self.vaultsDebtDistribution.totalSharesD18;
+            uint256 vaultSharesD18 = self.vaultsDebtDistribution.getActorShares(actorId);
+
+            uint256 numRewards = self.rewardIds.length();
+            for (uint256 i = 0; i < numRewards; i++) {
```

Comment:
As `updateRewardsToVaults` gets called for any vault "action" this is going to increase gas by a bit for every vault added to the pool? So the more collaterals we have the more and more gas we are going to consume for every delegation change

Reply:
No the loop is on number of RDs in the pool, not number if collateral types/vaults in the pool. so the gas increase is caused by the addition of any pool level RDs, which is inevitable considering this is the ask for the change

Reply:
Yeah sure. I meant to write rd not collateral.

Reply:
would be very good to see a gas impact for arbitrum and base deployments (with 4 and 2 distributors deployed on those).